docs/DeveloperGuide.md

```diff
@@ -541,20 +598,63 @@ testers are expected to do more *exploratory* testing.
+      Expected: First contact is deleted from the list. Details of the deleted contact are shown in the status message. 
 
    1. Test case: `delete 0`<br>
-      Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
+      Expected: No person is deleted. Error details shown in the status message. 
 
    1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+1. Deleting a person when the list is empty
+
+    1. Prerequisite: Ensure that there are no contacts in the app.
+
+    1. Test case: `delete 1`<br>
+       Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
+
+    1. Other incorrect delete commands to try: `delete`, `delete x` , `...` (where x can be any number)<br>
+       Expected: Similar to previous.
+    
 
 ### Saving data
 
-1. Dealing with missing/corrupted data files
+1. Dealing with missing data file
+
+    1. Navigate to ./data/addressbook.json and delete the addressbook.json file.
+   
+    1. Launch the app by double-clicking the jar file.<br>
+       Expected: The default list of contacts is loaded.
+
+1. Dealing with corrupted data file
+
+    1. Navigate to ./data/addressbook.json. Right click the addressbook.json file and open in TextEdit.
+
+    1. Delete all the contents of the file and type some symbols (e.g. `&*$@`).
+
+    1. Launch the app by double-clicking the jar file.<br>
+       Expected: The app loads with no contacts.
+                                       
+--------------------------------------------------------------------------------------------------------------------
+
+## **Appendix: Effort**   
+This project required substantial effort due to the complexity of expanding beyond the initial structure of AddressBook Level 3 (AB3).
+The following describes the challenges, difficulty level, and effort involved, as well as achievements attained through enhancements and
+additional features.
+
+### Difficulty Level and Challenges
+
+* **Additional Fields**: Our project manages additional fields such as Telegram handle, module, and contact type. Integrating these fields required modifying both data structures and UI components, as well as adapting backend logic to manage and validate these new attributes.
+* **Enhanced Search Functionality**: We implemented additional search functionalities, allowing users to find contacts by tags and Telegram handles, in addition to names. This required integration of the find command with additional fields.
+* **UI Enhancements**: Significant effort went into redesigning the user interface to make it aesthetically pleasing. In addition to improving layout and visual styling, we introduced a light and dark mode, ensuring each element was clearly visible in both themes.
```

Comment:
Would "we introduced both light and dark modes" sound better? 🤔

Reply:
Changed!